### DISCUSSION TITLE
first-line-indent does not work

### DISCUSSION THREAD

**Discussion by retsyo** [typst/typst#1011] [General]
as the following source show, I have used `#set par(first-line-indent: 2em, justify: false)`, but the first paragraph does not indent as paragraph 2 does. so what is the problem? Thanks

<pre>

#set heading(numbering: "1.")

#set text(
  font: "New Computer Modern",
  size: 10pt
)

#set page(
  paper: "a4",
  margin: (x: 1.8cm, y: 1.5cm),
)

#set par(first-line-indent: 2em, justify: false)
#show par: set block(spacing: 0.65em)

这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。

    这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。这是一段汉字测试。
</pre>

![bug](https://user-images.githubusercontent.com/7960913/234887230-09261e20-b03a-4bc8-b0e9-cb4c49c9d649.png)

**Comment by Enivex** [comment]
This is intentional

https://typst.app/docs/reference/layout/par/

> Only the first line of a consecutive paragraph will be indented (not the first one in a block or on the page).

**Reply by retsyo** [reply]
:(
For serious Chinese document, paragrahs always has a 2em indentation in the first line.

**Comment by jonathan-hellwig** [comment]
I am also interested in this. For IEEE style papers the first paragraph needs to be indented. Do you know of any workarounds?

**Reply by N3xed** [reply]
You can always create manual indentation with `#h(2em)` ([docs](https://typst.app/docs/reference/layout/h/)).
